src/common/helpers/validation-warnings.js

```diff
@@ -283,22 +264,58 @@ export const generatePopComponentWarnings = (payload) => {
     const components = wasteItem.pops.components
 
     // Check if source is one of the values that expects components
-    if (
-      sourceOfComponents &&
-      sourceOfComponents !== POP_COMPONENT_SOURCES.NOT_PROVIDED &&
-      (!components || components.length === 0)
-    ) {
+    if (isPopComponentsEmpty(sourceOfComponents, components)) {
+      warnings.push({
+        key: `wasteItems[${index}].pops.components`,
+        errorType: VALIDATION_ERROR_TYPES.NOT_PROVIDED,
+        message: `POP components are recommended when source of components is one of ${Object.values(sourceOfComponentsProvided).join(', ')}`
+      })
+    }
+
+    if (isPopConcentrationMissing(sourceOfComponents, components)) {
       warnings.push({
         key: `wasteItems[${index}].pops.components`,
         errorType: VALIDATION_ERROR_TYPES.NOT_PROVIDED,
-        message: `POP components are recommended when source of components is ${sourceOfComponents}`
+        message: `POP concentration is recommended when source of components is one of ${Object.values(sourceOfComponentsProvided).join(', ')}`
       })
     }
   })
 
   return warnings
 }
 
+/**
+ * Determines if POP components is an empty array
+ * @param {Object} sourceOfComponents - The POP source of components
+ * @param {Object} components - The POP components
+ * @returns {Boolean} True if POP components array is empty, otherwisse false
+ */
+function isPopComponentsEmpty(sourceOfComponents, components) {
+  return (
+    sourceOfComponents &&
+    sourceOfComponents !== 'NOT_PROVIDED' &&
+    Array.isArray(components) &&
+    components.length === 0
+  )
+}
+
+/**
+ * Determines if any of the POP components has a missing concentration value
+ * @param {Object} sourceOfComponents - The POP source of components
+ * @param {Object} components - The POP components
+ * @returns {Boolean} True if any of the POP components has a missing concentration value, otherwise false
+ */
+function isPopConcentrationMissing(sourceOfComponents, components) {
```

Comment:
is it possible for a payload to omit components altogether like { pops: { containsPops: true, sourceOfComponents: 'GUIDANCE' } }
if so the validator crashes instead of returning warnings